File: src/JavaScript/CreationalPatterns/AbstractFactory/Furniture/concreteFactories/contemporaryFurnitureFactory.js

const IFurnitureFactory = require("../interfaces/furnitureFactory");
const ContemporaryCabinet = require("../products/contemporaryCabinet");
const ContemporaryChair = require("../products/contemporaryChair");
const ContemporaryDiningTable = require("../products/contemporaryDiningTable");

/**
 * @class
 * @implements {IFurnitureFactory}
 */
class ContemporaryFurnitureFactory {
    /**
     * @override
     */
    createCabinet() {
        return new ContemporaryCabinet();
    }

    /**
     * @override
     */
    createChair() {
        return new ContemporaryChair();
    }

    /**
     * @override
     */
    createDiningTable() {
        return new ContemporaryDiningTable();
    }
}

module.exports = ContemporaryFurnitureFactory;

File: src/C#/CreationalPatterns/AbstractFactory/Furniture/ConcreteFactories/ContemporaryFurnitureFactory.cs
using Furniture.Interfaces;
using Furniture.Products;
using System;
using System.Collections.Generic;
using System.Text;

namespace Furniture.ConcreteFactories
{
    public class ContemporaryFurnitureFactory : IFurnitureFactory
    {
        public IFurniture CreateCabinet()
        {
            return new ContemporaryCabinet();
        }

        public IFurniture CreateChair()
        {
            return new ContemporaryChair();
        }

        public IFurniture CreateDiningTable()
        {
            return new ContemporaryDiningTable();
        }
    }
}
